st2-ask/example_workflow.yaml

```diff
@@ -0,0 +1,128 @@
+        #
+        # Note that we're providing our own JSON schema to the `schema` parameter here. `st2.ask` will
+        # use this to validate data stored in the execution context instead of the default schema that
+        # validates a simple approve/reject response from the user.
+        # When we do this, the action doesn't have a way of knowing what a "rejected" approval means,
+        # so this will block in `pending` status until someone sends data into the context that will validate
+        # the schema.
+        # In this event, the user may want to look into writing their own action that has more detailed checks of
+        # the data once validated, if they need this.
+        ask_for_2fa_token:
+            action: st2.ask
+            input:
+                schema:
+                    title: approval_data
+                    type: object
+                    properties:
+                        second_factor:
+                           type: string
+                           description: Please enter passcode for the second factor authentication
+                           secret: true
+                    required: ["second_factor"]
+            publish:
+                second_factor: <% task(ask_for_2fa_token).result.second_factor %>
+            on-success:
+                - send_request_with_2fa
+
+        # Here, we're using the "second_factor" variable published in the previous st2.ask action as input to this action.
+        # This demonstrates not only the ability for st2.ask to pause a workflow, but to retrieve information at runtime
+        # that's required for a future action.
+        send_request_with_2fa:
+            action: testpack.request_with_2fa
+            input:
+                username: foo
+                password: bar
+                token: <% $.second_factor %>
+            on-success:
+                - multi_stage_approval
+
+        # This final st2.ask example illustrates how multiple responders can be used to satisfy an st2.ask action.
+        # In this case, bob and fred would provide their response via Web UI, CLI, or API, and each value would be passed into the execution
+        # asynchronously. Once all of the values are present, the execution will resume.
+        multi_stage_approval:
+            action: st2.ask
+            input:
+                schema:
+                    title: approval_data
+                    type: object
+                    properties:
+                        bob_response:
+                           type: string
+                           description: Please enter bob's token
+                           secret: true
+                        fred_response:
+                           type: string
+                           description: Please enter fred's token
+                           secret: true
+                    required: ["bob_response", "fred_response"]
```

Comment:
This won't work. If we use st2 response, it'll ask for both Bob and Fred's responses. If we think about it, is Bob going to respond for Fred using st2 response? or Fred going to answer for Bob? They log in and response separately for themselves. We'll need two separate st2.ask execution with one notifying Bob and another notifying Fred.